screens/TournamentDetails/index.tsx

```diff
@@ -144,6 +154,15 @@ export default function TournamentDetailsScreen({
 
     </View>
   );
+
+  if (viewMode === "join") {
+    return (
+      <JoinDetails tournament={tournament} onSubmitJoin={(bracket: Bracket) => submitJoinBracket(bracket)} />
```

Comment:
noice

Reply:
is there a reason to pass onSubmitJoin down as a prop?

Reply:
is it for the spinner

Reply:
Yeah, this way the request and loading state is handled on the TournamentDetails component rather than down in the JoinDetails component